src/Kiota.Builder/Refiners/ShellRefiner.cs

```diff
@@ -0,0 +1,258 @@
+                                                .OfType<CodeProperty>()
+                                                .FirstOrDefault(x => x.Name.Equals(currentClass.Name, StringComparison.OrdinalIgnoreCase));
+                if (sameNameProperty != null)
+                {
+                    currentClass.RemoveChildElement(sameNameProperty);
+                    sameNameProperty.SerializationName ??= sameNameProperty.Name;
+                    sameNameProperty.Name = $"{sameNameProperty.Name}_prop";
+                    currentClass.AddProperty(sameNameProperty);
+                }
+            }
+            CrawlTree(currentElement, DisambiguatePropertiesWithClassNames);
+        }
+        private static void MakeEnumPropertiesNullable(CodeElement currentElement)
+        {
+            if (currentElement is CodeClass currentClass && currentClass.IsOfKind(CodeClassKind.Model))
+                currentClass.GetChildElements(true)
+                            .OfType<CodeProperty>()
+                            .Where(x => x.Type is CodeType propType && propType.TypeDefinition is CodeEnum)
+                            .ToList()
+                            .ForEach(x => x.Type.IsNullable = true);
+            CrawlTree(currentElement, MakeEnumPropertiesNullable);
+        }
+        private static void RemoveModelClasses(CodeElement currentElement)
+        {
+            if (currentElement is CodeClass currentClass && currentClass.IsOfKind(CodeClassKind.Model))
+            {
+                var codeNamespace = currentClass.Parent as CodeNamespace;
+                codeNamespace.RemoveChildElement(currentClass);
+            }
+            CrawlTree(currentElement, RemoveModelClasses);
+        }
+
+        private static void RemoveEnums(CodeElement currentElement)
+        {
+            if (currentElement is CodeEnum currentEnum)
+            {
+                var codeNamespace = currentElement.Parent as CodeNamespace;
+                codeNamespace.RemoveChildElement(currentEnum);
+            }
+            CrawlTree(currentElement, RemoveEnums);
+        }
+
+        private static void RemoveConstructors(CodeElement currentElement)
+        {
+            if (currentElement is CodeMethod currentMethod && currentMethod.IsOfKind(CodeMethodKind.Constructor))
+            {
+                var codeClass = currentElement.Parent as CodeClass;
+                codeClass.RemoveChildElement(currentMethod);
+            }
+            CrawlTree(currentElement, RemoveConstructors);
+        }
+
+        private static void CreateCommandBuilders(CodeElement currentElement)
+        {
+            if (currentElement is CodeClass currentClass && currentClass.IsOfKind(CodeClassKind.RequestBuilder))
+            {
+                // Replace Nav Properties with BuildXXXCommand methods
+                var navProperties = currentClass.GetChildElements().OfType<CodeProperty>().Where(e => e.IsOfKind(CodePropertyKind.RequestBuilder));
```

Comment:
You should have an "OfKind" extension method for IEnumerable<CodeProperty>

Reply:
The `OfKind` function seems to return only one item in the existing code. I think implementing it and returning a collection will be confusing, so I'll create a new function `AllOfKind`.

Reply:
yeah we have a couple of methods like that across code (method|parameter|class)(collection)? and they are not aligned, if you don't mind doing some cleanup so we have OfKind -> single result and AllOfKind -> IEnumerable, it'd be appreciated.